build-coordinator/src/main/java/org/jboss/pnc/coordinator/builder/DefaultBuildCoordinator.java

```diff
@@ -147,8 +151,8 @@ public BuildSetTask build(BuildConfigurationSet buildConfigurationSet, User user
                 rebuildAll,
                 getBuildStatusChangedEventNotifier(),
                 () -> datastoreAdapter.getNextBuildRecordId());
-        BuildSetStatusChangedEvent event = buildSetTask.setStatus(BuildSetStatus.NEW);
-        buildSetStatusChangedEventNotifier.fire(event);
+        notifyBuildSetStatusChanged(buildSetTask, BuildSetStatus.NEW);
```

Comment:
I don't understand. Why does it have to return anything? BuildTask is mutable...

if status desc is always set, I'd add it too. Then we have all things that happen when status is changed in one place and we're sure we won't forget to fix some if there's some bug in it.